src/all/globalcomix/src/eu/kanade/tachiyomi/extension/all/globalcomix/GlobalComix.kt

```diff
@@ -0,0 +1,265 @@
+            // The results property will be a JSON array
+            response.parseAs<MangasDto>().payload!!.let { dto ->
+                MangasPage(
+                    dto.results.map { it -> it.createManga() },
+                    dto.pagination.hasNextPage,
+                )
+            }
+        } else {
+            // However, when using the 'id:' query prefix (via the UrlActivity for example),
+            // the response is a single manga and the results property will be a JSON object
+            MangasPage(
+                listOf(
+                    response.parseAs<MangaDto>().payload!!
+                        .results
+                        .createManga(),
+                ),
+                false,
+            )
+        }
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        // If the query is a slug ID, return the manga directly
+        if (query.startsWith(prefixIdSearch)) {
+            val mangaSlugId = query.removePrefix(prefixIdSearch)
+
+            if (mangaSlugId.isEmpty()) {
+                throw Exception(intl["invalid_manga_id"])
+            }
+
+            val url = apiMangaUrl.toHttpUrl().newBuilder()
+                .addPathSegment(mangaSlugId)
+                .build()
+
+            return GET(url, headers, CacheControl.FORCE_NETWORK)
+        }
+
+        return simpleQueryRequest(page, orderBy = "relevance", query)
+    }
+
+    override fun searchMangaParse(response: Response): MangasPage = popularMangaParse(response)
+
+    override fun getMangaUrl(manga: SManga): String = "$webComicUrl/${titleToSlug(manga.title)}"
+
+    override fun mangaDetailsRequest(manga: SManga): Request {
+        val url = apiMangaUrl.toHttpUrl().newBuilder()
+            .addPathSegment(titleToSlug(manga.title))
+            .build()
+
+        return GET(url, headers, CacheControl.FORCE_NETWORK)
+    }
+
+    override fun mangaDetailsParse(response: Response): SManga =
+        response.parseAs<MangaDto>().payload!!
+            .results
+            .createManga()
+
+    override fun chapterListRequest(manga: SManga): Request {
+        val url = apiSearchUrl.toHttpUrl().newBuilder()
+            .addPathSegment(manga.url) // manga.url contains the the comic id
+            .addPathSegment("releases")
+            .addQueryParameter("lang_id", extLang)
+            .addQueryParameter("all", "true")
+            .toString()
+
+        return GET(url, headers, CacheControl.FORCE_NETWORK)
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        if (response.code == 204) {
+            return emptyList()
+        }
+
+        return response.parseAs<ChaptersDto>().payload!!.results.mapNotNull { dto ->
+            if (dto.isPremium && !preferences.showLockedChapters) {
+                null
+            } else {
+                dto.createChapter()
+            }
+        }
```

Comment:
```suggestion
        return response.parseAs<ChaptersDto>().payload!!.results.filterNot { dto ->
            dto.isPremium && !preferences.showLockedChapters
        }.map { it.createChapter() }
```
may need to adjust for lint